java/src/main/java/ai/rapids/cudf/WindowOptions.java

```diff
@@ -158,21 +187,58 @@ public Builder window(ColumnVector precedingCol, ColumnVector followingCol) {
       return this;
     }
 
+    /**
+     * Set the size of the range window
```

Comment:
We should document that the scalar values passed in need to out live the WindowOptions passed in. `WindowOptions` is not `AutoCloseable`. 

I think I would prefer it if we wanted to make `WindowOptions` `AutoCloseable` too, and then have it increment the reference count on the Scalars when it gets them. That would be more of  breaking change, but it would make working with the API a lot cleaner.

Reply:
Done. A lot of changes for tests.